### PR TITLE
Set up Continuous Integration / Docker

### PR DESCRIPTION
**Pull request by sjmatta**
* Requires a `settings-production.json` in the root directory
* Currently exposes the Meteor application directly, without a reverse proxy
* Added native bcrypt install - this may be more appropriately placed in `package.json`
* Looking for additional feedback

### CODE REVIEW

**Comment by sjmatta** [comment]
* Add `circle.yml` to build Docker container and archive as an artifact
* (Also archives docker-compose config for convenience)
* Testing is not part of continuous integration, yet

**Comment by sjmatta** [comment]
Finishes #120003861

**Comment by sjmatta** [comment]
* Tests are now run
* All tests are currently passing, but, I can't figure out why Chimp (or, at least, run_tests.sh) isn't returning a non-zero code when tests fail

**Comment by sjmatta** [comment]
Unfortunately, the build isn't failing when tests fail. I'll work on this some more if I have additional time, but I think given that as known technical debt, this is ready to merge.

**Comment by nathanathan** [comment]
Could you create a separate Dockerfile for CI and testing. The one you've been editing is currently used for deployments, so I don't want to merge all of these changes to it since some will probably break our deployment process.

@frosario Could you comment on this as well.

File: Dockerfile [hunk 1]
```diff
@@ -1,38 +1,28 @@
-FROM ubuntu:14.04.4
+FROM ubuntu:14.04
```
Comment (frosario) [inline Dockerfile]: Why base off an older obsolete release of ubuntu?

File: Dockerfile [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
 
-# Replace shell with bash so we can source files
-RUN rm /bin/sh && ln -s /bin/bash /bin/sh
```
Comment (frosario) [inline Dockerfile]: This line guarantees that when you run an exec command, that we are using the bash shell and not the minimal dash shell. Bash is necessary for things like properly sourcing in environment vars.

File: Dockerfile (hunk 1 above)
Comment (sjmatta) [inline Dockerfile]: According to https://hub.docker.com/_/ubuntu/, :14.04 will always point to the latest release of ubuntu 14.04 (currently it points to 14.04.5).

File: Dockerfile [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
-
-# Install apt package dependencies
-RUN apt-get clean all && apt-get update && \
-    apt-get -y install wget curl python make g++ git supervisor && \
```
Comment (frosario) [inline Dockerfile]: We need supervisor as the process that keeps the container alive always. Typical docker containers are supposed to be single process, but we had to take a very hybrid approach for our stakeholder's needs.

File: Dockerfile (hunk 1 above)
Comment (sjmatta) [inline Dockerfile]: Oh, I may have misunderstood. You mean to use 16?

File: Dockerfile (hunk 3 above)
Comment (sjmatta) [inline Dockerfile]: @nathanathan has asked I make this Dockerfile separate from the existing one, for use in dev. This one won't be used to create containers delivered to your stakeholders.

File: Dockerfile (hunk 1 above)
Comment (frosario) [inline Dockerfile]: Correct. 16 has been out for awhile now and seems pretty stable. Any reason to stay on 14?

File: Dockerfile (hunk 3 above)
Comment (frosario) [inline Dockerfile]: Ok sounds good.

**Comment by nathanathan** [comment]
I'm closing this and recreating the changes in new files on another branch.

**Pull request closed without merging**